### PR TITLE
DS-2704

### PR DESCRIPTION
**Pull request by jrwishart**

### CODE REVIEW

File: R/regression.R [hunk 1]
```diff
@@ -618,46 +637,102 @@ importanceFooter <- function(x)
...
+    model <- fitted.model$model
+    .design <- fitted.model$design
+    .estimation.data <- fitted.model$estimation.data
+    .formula <- fitted.model$formula
+    remove.outliers <- checkAutomaterOutlierRemovalSetting(outlier.prop.to.remove)
+    if (remove.outliers)
+    {
+        # Don't support Multinomial Logit for now.
+        if (type == "Multinomial Logit")
+            stop("Automated outlier removal and re-fitting a 'Multinomial Logit' model is not supported")
```
Comment (mwmclean) [inline R/regression.R]: what about instead warning and continuing without the outlier removal in the multinomial case?

File: tests/testthat/test-_bugs.R [hunk 2]
```diff
@@ -86,12 +86,12 @@ test_that("Duplicated labels",
             Q5r_2 = structure(c(2, 2, 4, 5, 5, 5, 4, 5, 5, 3, 5, 3, 5,
             4, 3, 4, 3, 4, 5, 5), questiontype = "NumberMulti", name = "Q5r_2", label = "Landscaping", question = "How satisfied were you with... 2")), row.names = c(NA,
         20L), class = "data.frame")
-
-    expect_warning(print(Regression(Q4_2~.,, data = df.small, show.labels = TRUE,
+    test.formula <- as.formula(paste0("Q4_2 ~ ", paste0(names(df.small)[-1], collapse = " + ")))
```
Comment (mwmclean) [inline tests/testthat/test-_bugs.R]: Why was the `.` removed from these tests?

File: tests/testthat/test-multipleimputation.R [hunk 3]
```diff
@@ -43,90 +43,92 @@ test_that("Multiple imputation with auxiliary variables ", {
 test_that("Multiple imputation run using Regression", {
     est <- flipData::EstimationData(Overall ~ Fees + Interest + Phone + Branch + Online + ATM + rnd + rnd1 + rnd2 + rnd3 + rnd4, zbank, missing = "Multiple imputation", m = 10)
     models <- lapply(est$estimation.data, FUN = function(x) Regression(Overall ~ Fees + Interest + Phone + Branch + Online + ATM + rnd + rnd1 + rnd2 + rnd3 + rnd4, data = x))
-    coefs <- MultipleImputationCoefficientTable(models)
+    coefs <- expect_error(MultipleImputationCoefficientTable(models), NA)
```
Comment (mwmclean) [inline tests/testthat/test-multipleimputation.R]: can remove `coefs <-`

File: R/regression.R [hunk 4]
```diff
@@ -62,6 +64,10 @@
 #' @param recursive.call Used internally to indicate if call is a result of recursion (e.g., multiple imputation).
 #' @param effects.format A list of items \code{max.label} (the maximum length of a factor label on the x-axis)
 #'   and \code{y.title} (the title of the y-axis, defaults to outcome label).
+#' @param outlier.prop.to.remove A single numeric value that determines the percentage of data points to remove from the
+#'   analysis. The data points removed correspond to those in the proportion with the largest residuals.
+#'   A value of 0 or NULL would denote no points are removed. A value x, with 0 < x < 0.5 (not inclusive) would
+#'   denote that a percentage between none and 50\% of the data points are removed.
 #' @param ... Additional argments to be passed to  \code{\link{lm}} or, if the
 #'  data is weighted,  \code{\link[survey]{svyglm}} or \code{\link[survey]{svyolr}}.
 #' @details In the case of Ordered Logistic regression, this function computes a proporional odds model using
```
Comment (mwmclean) [inline R/regression.R:73]: Would be good to document how the outlier detection works in the details and discussed the types of residuals used in the various cases.

**Review by mwmclean — COMMENTED** [review]
image
Unfortunately, seeing a bunch of errors on my machine when running locally.

File: tests/testthat/test-_bugs.R (hunk 2 above)
Comment (jrwishart) [reply tests/testthat/test-_bugs.R]: I don't recall exactly why I did that for this case. I did go through a lot of test files and tidied up some things. It doesn't look necessary in this case and the tests still pass when the `.` is restored in the formula. Code restored in https://github.com/Displayr/flipRegression/commit/4d50650a1ac03dd71c97b804f034445ad1e0bedc

File: R/regression.R (hunk 1 above)
Comment (jrwishart) [reply R/regression.R]: I agree that is better. Code amended to do this in https://github.com/Displayr/flipRegression/commit/4d50650a1ac03dd71c97b804f034445ad1e0bedc

File: R/regression.R (hunk 4 above)
Comment (jrwishart) [reply R/regression.R:73]: Added more details in the documentation in https://github.com/Displayr/flipRegression/commit/4d50650a1ac03dd71c97b804f034445ad1e0bedc. Let me know if more details are required.

File: tests/testthat/test-multipleimputation.R (hunk 3 above)
Comment (jrwishart) [reply tests/testthat/test-multipleimputation.R]: Removed in https://github.com/Displayr/flipRegression/commit/4d50650a1ac03dd71c97b804f034445ad1e0bedc

**Comment by jrwishart** [comment]
Locally I still get all tests passing and the travis tests pass too. The only discrepancy I have is an RNG issue from testing the entire package (i.e. `devtools::test()`) results in 8 errors in https://github.com/Displayr/flipRegression/blob/750ae2e99b8b8643d1c55d65da620377beff21d0/tests/testthat/test-regression.R#L25-L48. However those same lines pass when the file is tested individually (i.e. `devtools::test_file("tests/testthat/test-regression.R")`. Those lines were changed in https://github.com/Displayr/flipRegression/commit/59d53685eeb09fed29d0e6a8fd20b64695c0c390 Those 8 tests pass on travis.

**Pull request merged**